src/block-management/block-manager.js

```diff
@@ -52,7 +53,7 @@ export default class BlockManager extends React.Component<PropsType, StateType>
 			inspectBlocks: false,
 			blockTypePickerVisible: false,
 			selectedBlockType: 'core/paragraph', // just any valid type to start from
-			html: '', // This is used to hold the Android html text input state.
+			html: '',
```

Comment:
> My original idea was to let HTMLInputView handle just text

It was a perfect start, nothing wrong there. It just turns out that html related code/logic is found in many places in `block-manager.js` while the block manager ideally has nothing to do with the raw html. So, it now makes sense to try to abstract all html handling away.

> even though it would create a coupling (probably not an issue right now?)

Hmm, not sure which coupling you have in mind. Coupling to the a block's list? The block list is currently the master source of info and is stored in Redux so, ideally we could even have the HTMLTextInput be completely separated from the BlockManager, to get rendered by the `MainScreen` if the `showHtml` prop/state is true. In a sense, I think I see a path to completely decouple the html editing from the block list itself that way. Does that make sense?

All in all, it's not a high priority to go down that route now so, not going to block this PR on this one.

> We would have to generate the block list on every key stroke and give it back via props function

Ah, I see what you mean. Sounds like a case where we would only want to update the Redux state via a debouncer or when HTMLTextInput gets blurred. I wonder how GB-web does it with its full post html editor.

> We hold a reference to the component and call a function on it to get the new set of blocks (or the html text itself). It will probably work but I believe it's not too react-like.

Yeah, woudn't like us to end up there if we can avoid it.

All in all, I'd say let's timebox this one. Can't think of great benefit optimizing the code separation too much right now.

Reply:
> Ideally we could even have the HTMLTextInput be completely separated from the BlockManager, to get rendered by the MainScreen.

I see. That sounds like a great plan!
If it was the plan all along, I'm sorry I missed it 😆

I see that my approach wouldn't work, but you gave so many tips that I will try again with this new approach. 

I'll keep you posted. 😁